### PR TITLE
Error handling and logging improvements

### PR DESCRIPTION
**Pull request by glennmatthews**
* Improve the wording and formatting of various job log messages
* Improve listing of configured providers in notificationsource detail view
* Avoid erroring out when encountering a Provider that has a null `emails_circuit_maintenances` custom field value
* Handle case where `emails_circuit_maintenances` comma-separated list contains whitespace

### CODE REVIEW

**Review by dgarros — APPROVED** [review]
LGTM

File: nautobot_circuit_maintenance/handle_notifications/sources.py [hunk 1]
```diff
@@ -253,8 +254,10 @@ def extract_provider_data_types(email_source: str) -> Tuple[str, str, str]:
             )
         """
         for provider in Provider.objects.all():
-            if "emails_circuit_maintenances" in provider.custom_field_data:
-                if email_source in provider.custom_field_data["emails_circuit_maintenances"].split(","):
+            if provider.custom_field_data.get("emails_circuit_maintenances"):
```
Comment (itdependsnetworks) [inline nautobot_circuit_maintenance/handle_notifications/sources.py:257]: Why not in single if conditional?

File: nautobot_circuit_maintenance/tests/test_sources.py [hunk 2]
```diff
@@ -128,7 +128,7 @@ def test_extract_provider_data_types_no_provider_parser(self):
             (
                 "",
                 "",
-                f"Unexpected provider {provider_type} received from {email_source}, so not getting the notification",
+                f"{email_source} is for provider type `{provider_type}`, not presently supported",
```
Comment (itdependsnetworks) [inline nautobot_circuit_maintenance/tests/test_sources.py:131]:
```suggestion
                f"{email_source} is for provider type `{provider_type}`, which is not presently supported",
```

File: nautobot_circuit_maintenance/tests/test_sources.py [hunk 3]
```diff
@@ -152,7 +152,7 @@ def test_extract_provider_data_types_ok(self, provider_type, data_types, error_m
             (
                 data_types if not error_message else "",
                 provider_type if not error_message else "",
-                f"Unexpected provider unknown received from {email_source}, so not getting the notification"
+                f"{email_source} is for provider type `{provider_type}`, not presently supported"
```
Comment (itdependsnetworks) [inline nautobot_circuit_maintenance/tests/test_sources.py:155]:
```suggestion
                f"{email_source} is for provider type `{provider_type}`, which is not presently supported"
```

File: nautobot_circuit_maintenance/handle_notifications/sources.py (hunk 1 above)
Comment (glennmatthews) [reply nautobot_circuit_maintenance/handle_notifications/sources.py:257]: You mean:

```python
             if (
                 provider.custom_field_data.get("emails_circuit_maintenances")
                 and email_source in [
                    source.strip() for source in provider.custom_field_data["emails_circuit_maintenances"].split(",")
                ]
            ):
```

Feels a bit unwieldy to me but I'm not opposed to it.

Comment (itdependsnetworks) [reply nautobot_circuit_maintenance/handle_notifications/sources.py:257]: i won't lose sleep over it and only see pros/cons. If I see dual if statements I am confused, conversely if I see complex if statement its hard to read.

Also thinking to invert the truth table to keep it less nested.

```
            if provider.custom_field_data.get("emails_circuit_maintenances") is None:
               continue
            if email_source in [
                    source.strip() for source in provider.custom_field_data["emails_circuit_maintenances"].split(",")
                ]:
                provider_type = provider.slug
                break
```

**Comment by glennmatthews** [comment]
Addressed review comments - thanks! Some additional fixes and enhancements:

- Clarified some aspects of README.md
- Fixed an issue where we weren't recognizing email addresses containing `-` characters as valid
- Added `source_header` configuration option to support a case where messages are forwarded from a mailing list and so we need to inspect the `X-Original-From` header instead of the `From` header to identify the actual message source.

File: nautobot_circuit_maintenance/handle_notifications/sources.py [hunk 4]
```diff
@@ -220,6 +221,7 @@ def validate_providers(self, job_logger: Job, notification_source: NotificationS
             )
             return False
 
+        # job_logger.log_debug(message=f"Fetching emails from {self.emails_to_fetch}")
```
Comment (chadell) [inline nautobot_circuit_maintenance/handle_notifications/sources.py:224]: is this left behind  for easy debugging when needed?

**Review by chadell — COMMENTED** [review]
nice improvements!
My only concern is about the scope of `source_header` because it looks to me as provider specific, so the same instance of the Plugin could be parsing emails from a provider that uses the "From" and other that uses the "X-Original-From", and from the implementation we are only covering on or another.
So, naturally it comes to me to see this as an extra optional custom field for the Provider, that would use "From" if not specified.
And, as you already pointed out on the PR, we should take this into account when fetching emails from a Source, using the "source_header".

File: nautobot_circuit_maintenance/handle_notifications/sources.py (hunk 4 above)
Comment (glennmatthews) [reply nautobot_circuit_maintenance/handle_notifications/sources.py:224]: Yeah - it was useful in troubleshooting why some emails weren't getting fetched, but is a bit noisy during normal operation. 

I see some other plugins have added an explicit "Debug" or "Verbose" checkbox to their Jobs that can be toggled to get more detailed logs when desired - maybe something to consider adding here?

**Comment by glennmatthews** [comment]
I agree that the `source_header` should probably be a provider-specific configuration - I'll look into fixing that as soon as I can find the time.

File: nautobot_circuit_maintenance/handle_notifications/sources.py (hunk 4 above)
Comment (chadell) [reply nautobot_circuit_maintenance/handle_notifications/sources.py:227]: yes, it would be great

**Review by chadell — APPROVED** [review]
LGTM

**Pull request merged**